File: src/microscript/Parameter.cs
/**
 * MicroScript — The programming language
 * Copyright (c) 2024-2025 Cyril John Magayaga
 * 
 * It was originally written in C# programming language.
 */
namespace com.magayaga.microscript
{
    public class Parameter
    {
        private readonly string name;
        private readonly string type;

        public Parameter(string name, string type)
        {
            this.name = name;
            this.type = type;
        }

        public string GetName()
        {
            return name;
        }

        public new string GetType()
        {
            return type;
        }
    }
}

File: src/parameter.go
/**
 * MicroScript — The programming language
 * Copyright (c) 2024, 2025 Cyril John Magayaga
 *
 * It was originally written in Go programming language
 */
package main

// Parameter represents a function parameter
type Parameter struct {
	Name string
	Type string
}

// NewParameter creates a new parameter
func NewParameter(name, paramType string) *Parameter {
	return &Parameter{
		Name: name,
		Type: paramType,
	}
}

// GetName returns the parameter name
func (p *Parameter) GetName() string {
	return p.Name
}

// GetType returns the parameter type
func (p *Parameter) GetType() string {
	return p.Type
}
